backend/api/Services/DocumentService.cs

```diff
@@ -16,63 +16,65 @@ namespace Pims.Api.Services
         {
-            Task<ExternalResult<QueryResult<DocumentMetadata>>> task = documentRepository.GetDocumentMetadataAsync(documentId, ordering, page, pageSize);
-            task.Wait();
-            return task.Result;
+            ExternalResult<QueryResult<DocumentType>> result = await documentStorageRepository.GetDocumentTypesAsync(ordering, page, pageSize);
+            return result;
         }
 
-        public async Task<ExternalResult<FileDownload>> DownloadFileAsync(int documentId, int fileId)
+        public async Task<ExternalResult<QueryResult<DocumentDetail>>> GetStorageDocumentList(string ordering = "", int? page = null, int? pageSize = null)
         {
-            ExternalResult<FileDownload> downloadResult = await documentRepository.DownloadFileAsync(documentId, fileId);
-            return downloadResult;
+            ExternalResult<QueryResult<DocumentDetail>> result = await documentStorageRepository.GetDocumentsListAsync(ordering, page, pageSize);
+            return result;
+        }
+
+        public async Task<ExternalResult<QueryResult<DocumentMetadata>>> GetStorageDocumentMetadata(long mayanDocumentId, string ordering = "", int? page = null, int? pageSize = null)
+        {
+            ExternalResult<QueryResult<DocumentMetadata>> result = await documentStorageRepository.GetDocumentMetadataAsync(mayanDocumentId, ordering, page, pageSize);
+            return result;
         }
 
-        public async Task<ExternalResult<FileDownload>> DownloadFileLatestAsync(int documentId)
+        public async Task<ExternalResult<FileDownload>> DownloadFileAsync(long mayanDocumentId, long mayanFileId)
         {
-            ExternalResult<DocumentDetail> documentResult = await documentRepository.GetDocumentAsync(documentId);
-            ExternalResult<FileDownload> downloadResult = await documentRepository.DownloadFileAsync(documentResult.Payload.Id, documentResult.Payload.FileLatest.Id);
+            ExternalResult<FileDownload> downloadResult = await documentStorageRepository.DownloadFileAsync(mayanDocumentId, mayanFileId);
             return downloadResult;
         }
 
-        public async Task<ExternalResult<DocumentDetail>> UploadDocumentAsync(int documentType, IFormFile fileRaw)
+        public async Task<ExternalResult<FileDownload>> DownloadFileLatestAsync(long mayanDocumentId)
         {
-            await this.avService.ScanAsync(fileRaw);
-            Task<ExternalResult<DocumentDetail>> task = documentRepository.UploadDocumentAsync(documentType, fileRaw);
-            task.Wait();
-            return task.Result;
+            ExternalResult<DocumentDetail> documentResult = await documentStorageRepository.GetDocumentAsync(mayanDocumentId);
+            ExternalResult<FileDownload> downloadResult = await documentStorageRepository.DownloadFileAsync(documentResult.Payload.Id, documentResult.Payload.FileLatest.Id);
```

Comment:
this is a pretty big '.' chain, how confident are we that this won't NPE? I'd rather we return a nicely formatted error if possible.

Reply:
fair call. I could add some checks to avoid crashes